src/snarky.d.ts

```diff
@@ -51,489 +53,806 @@ declare interface ProvablePure<T> extends Provable<T> {
+   * 
+   * @return A {@link Field} element equivalent to the multiplication of the {@link Field} element with itself.
    */
   square(): Field;
 
   /**
-   * Square roots this {@link Field} element.
+   * Take the square root of this {@link Field} element.
    *
-   * ```typescript
-   * x.square().sqrt().assertEquals(x);
+   * @example
+   * ```ts
+   * const someField = Field(42);
+   * const squareRoot = someField.sqrt();
+   * 
+   * squareRoot.mul(squareRoot).assertEquals(someField); // This statement is always true regardless of the value of `someField`
    * ```
+   * 
+   * **Warning**: This is a modular square root. Please see `div()` function for more details.
+   * 
+   * @return A {@link Field} element equivalent to the square root of the {@link Field} element.
    */
   sqrt(): Field;
 
   /**
-   * Serialize the {@link Field} to a string, e.g. for printing.
-   * This operation does NOT affect the circuit and can't be used to prove anything about the string representation of the Field.
+   * Serialize the {@link Field} to a string, e.g. for printing. If you try to print a {@link Field} without this function it will directly stringify the Field object, resulting in an unreadable output.
+   * 
+   * **Warning**: This operation does NOT affect the circuit and can't be used to prove anything about the string representation of the {@link Field}. Please use it only during debugging.
+   * 
+   * @example
+   * ```ts
+   * const someField = Field(42);
+   * console.log(someField.toString());
+   * ```
+   * 
+   * @return A string equivalent to the string representation of the Field.
    */
   toString(): string;
+
   /**
-   * Serialize this instance of a {@link Field} to a bigint.
-   * This operation does NOT affect the circuit and can't be used to prove anything about the bigint representation of the Field.
+   * Serialize the {@link Field} to a bigint, e.g. for printing. If you try to print a {@link Field} without this function it will directly stringify the Field object, resulting in an unreadable output.
+   * 
+   * **Warning**: This operation does NOT affect the circuit and can't be used to prove anything about the bigint representation of the {@link Field}. Please use it only during debugging.
+   * 
+   * @example
+   * ```ts
+   * const someField = Field(42);
+   * console.log(someField.toBigInt());
+   * ```
+   * 
+   * @return A bigint equivalent to the bigint representation of the Field.
    */
   toBigInt(): bigint;
-  /**
-   * Serialize this instance of a {@link Field} to a JSON string.
-   * This operation does NOT affect the circuit and can't be used to prove anything about the string representation of the Field.
-   */
-  toJSON(): string;
 
   /**
-   * Returns the size of this type.
+   * Serialize the {@link Field} to a JSON string, e.g. for printing. If you try to print a {@link Field} without this function it will directly stringify the Field object, resulting in an unreadable output.
```

Comment:
same as in https://github.com/o1-labs/snarkyjs/pull/917/files#diff-c986331433c1c055aa5d451d0b789e5f5e9aee97f754e6f1550fb09b5acc91b4R321